src/levenberg.jl

```diff
@@ -176,15 +186,29 @@ function SciMLBase.__init(prob::Union{NonlinearProblem{uType, iip},
     fu_tmp = zero(fu1)
     mat_tmp = zero(JᵀJ)
 
-    return LevenbergMarquardtCache{iip}(f, alg, u, fu1, fu2, du, p, uf, linsolve, J,
-        jac_cache, false, maxiters, internalnorm, ReturnCode.Default, abstol, prob, DᵀD,
+    tc = alg.termination_condition
+    mode = DiffEqBase.get_termination_mode(tc)
+
+    atol = _get_tolerance(abstol, tc.abstol, eltype(u))
+    rtol = _get_tolerance(reltol, tc.reltol, eltype(u))
+
+    storage = mode ∈ DiffEqBase.SAFE_TERMINATION_MODES ? NLSolveSafeTerminationResult() :
```

Comment:
Is this type stable?

Reply:
I believe so; the types are inferred in the cache struct. Something similar to want we did here: https://github.com/SciML/SimpleNonlinearSolve.jl/blob/21f14ff8e1bca2e20ee0de9158050a154ce3d585/src/broyden.jl#L53